src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/MethodTypeInference.cs

```diff
@@ -1965,6 +1968,27 @@ private bool LowerBoundInterfaceInference(TypeSymbol source, NamedTypeSymbol tar
             return true;
         }
 
+        internal static ImmutableArray<NamedTypeSymbol> ModuloNullabilityDifferences(IEnumerable<NamedTypeSymbol> interfaces, VarianceKind variance)
+        {
+            var dictionary = new Dictionary<NamedTypeSymbol, NamedTypeSymbol>(SymbolEqualityComparer.IgnoreNullableModifiersForReferenceTypes);
+
+            foreach (var @interface in interfaces)
+            {
+                if (dictionary.TryGetValue(@interface, out var found))
+                {
+                    dictionary.Remove(@interface);
+                    var merged = (NamedTypeSymbol)found.MergeNullability(@interface, variance);
```

Comment:
>found.MergeNullability(@interface, variance) [](start = 50, length = 44)

Can this operation fail in some way? Is result specified in that situation? Would we want to report any warnings if result is used?

Reply:
We'll have reported a warning at the source already (where duplicate implementations were introduced). The merging itself can always be done (`MergeNullableAnnotation` is fully defined).

---
In reply to: [321433393](https://github.com/dotnet/roslyn/pull/38460#discussion_r321433393) [](ancestors = 321433393)

Reply:
> We'll have reported a warning at the source already (where duplicate implementations were introduced). 

The type could come from a different assembly, no warnings would be reported anywhere.
The type can be defined in a different file and the warnings could be disabled there.

---
In reply to: [321458342](https://github.com/dotnet/roslyn/pull/38460#discussion_r321458342) [](ancestors = 321458342,321433393)

Reply:
My bad. Also, we don't report an error in the indirect cases. I'll see about reporting a warning then.

---
In reply to: [321466164](https://github.com/dotnet/roslyn/pull/38460#discussion_r321466164) [](ancestors = 321466164,321458342,321433393)

Reply:
From our discussion, it seems okay to not report a warning for dicey merges.

---
In reply to: [321467694](https://github.com/dotnet/roslyn/pull/38460#discussion_r321467694) [](ancestors = 321467694,321466164,321458342,321433393)